src/System.Private.CoreLib/shared/System/Number.NumberToDouble.cs

```diff
@@ -0,0 +1,450 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace System
+{
+    internal unsafe partial class Number
+    {
```

Comment:
Is there anything useful you are able to say about the algorithm used, for the benefit of anyone maintaining it?

Reply:
Possibly, but there wasn't any comments like that in the native code, so I can really only put what I am interpreting the code to be doing.

Given that the algorithm is currently known to be incorrect, I would think this should be replaced with a "correct" parsing algorithm soonish (hopefully) and we can add some proper algorithm comments then.